spec/classes/init_spec.rb

```diff
@@ -452,6 +482,26 @@
     end
   end
 
+  describe 'with service_provider parameter specified' do
+    context 'as a valid string' do
+      let(:params) { { :service_provider => 'myprovider' } }
+      let(:facts) { { :osfamily => 'Debian' } }
+
+      it { should contain_service('nscd_service').with({ 'provider' => 'myprovider' }) }
+    end
+
+    context 'as an invalid type' do
+      let(:params) { { :service_provider => ['not','a','string'] } }
+      let(:facts) { { :osfamily => 'Debian' } }
+
+      it 'should fail' do
+        expect {
+          should contain_class('nscd')
+        }.to raise_error(Puppet::Error)
```

Comment:
Could you please add the error message. A good way to find is to try

```ruby
}.to raise_error(Puppet::Error,/asdf/)
```

see the error and then update the test